File: 11.container-with-most-water.java
// @leet start
class Solution {
	public int maxArea(int[] height) {
		int l = 0;
		int r = height.length - 1;
		int answer = 0;
		while (l < r) {
			answer = Math.max(Math.min(height[l], height[r]) * (r - l), answer);
			if (height[r] < height[l]) {
				r--;
			} else {
				l++;
			}
		}
		return answer;
	}
}
// @leet end

File: 11.container-with-most-water.py
# @leet start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height) - 1
        answer = 0
        while l < r:
            answer = max(min(height[l], height[r]) * (r - l), answer)
            if height[r] < height[l]:
                r -= 1
            else:
                l += 1
        return answer


# @leet end
